# Project 3: Global Missing Migrants Dataset
Introduction: The Missing Migrants Project, initiated by the International Organization for Migration, documents migrants who have gone missing or have died in their migration to an another location. This project analyzes the data by region, incidents, migration route, and the time period from 2014-2023. 

In [49]:
import pandas as pd
df = pd.read_csv("Global Missing Migrants Dataset.csv")
df.head(5)

,Incident Type,Incident year,Reported Month,Region of Origin,Region of Incident,Country of Origin,Number of Dead,Minimum Estimated Number of Missing,Total Number of Dead and Missing,Number of Survivors,Number of Females,Number of Males,Number of Children,Cause of Death,Migration route,Location of death,Information Source,Coordinates,UNSD Geographical Grouping
0,Incident,2014,January,Central America,North America,Guatemala,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,"31.650259, -110.366453",Northern America
1,Incident,2014,January,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,0,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,"31.59713, -111.73756",Northern America
2,Incident,2014,January,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,0,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,"31.94026, -113.01125",Northern America
3,Incident,2014,January,Central America,North America,Mexico,1.0,0,1,0,0,1,0,Violence,US-Mexico border crossing,"near Douglas, Arizona, USA","Ministry of Foreign Affairs Mexico, Pima Count...","31.506777, -109.315632",Northern America
4,Incident,2014,January,Northern Africa,Europe,Sudan,1.0,0,1,2,0,1,0,Harsh environmental conditions / lack of adequ...,NaN,Border between Russia and Estonia,EUBusiness (Agence France-Presse),"59.1551, 28",Northern Europe


## Data Overview: 
A comprehensive record of missing migrants from the Missing Migrants project which was initiated by International Organization of Migration (IOM). This spans recorded missing migrants from 2014 to 2023. 

In [50]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13020 entries, 0 to 13019
Data columns (total 19 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Incident Type                        13020 non-null  object 
 1   Incident year                        13020 non-null  int64  
 2   Reported Month                       13020 non-null  object 
 3   Region of Origin                     12998 non-null  object 
 4   Region of Incident                   13020 non-null  object 
 5   Country of Origin                    13012 non-null  object 
 6   Number of Dead                       12470 non-null  float64
 7   Minimum Estimated Number of Missing  13020 non-null  int64  
 8   Total Number of Dead and Missing     13020 non-null  int64  
 9   Number of Survivors                  13020 non-null  int64  
 10  Number of Females                    13020 non-null  int64  
 11  Number of Males             

In [51]:
df.columns = df.columns.str.lower()
df.columns = df.columns.str.replace(' ', '_')

In [52]:
df.describe()

,incident_year,number_of_dead,minimum_estimated_number_of_missing,total_number_of_dead_and_missing,number_of_survivors,number_of_females,number_of_males,number_of_children
count,13020.000000,12470.000000,13020.000000,13020.000000,13020.000000,13020.000000,13020.000000,13020.000000
mean,2019.060753,2.623737,1.904378,4.417281,6.553763,0.458372,1.116052,0.238095
std,2.418738,9.728163,15.645297,19.950630,42.315444,2.853658,3.811134,2.417196
min,2014.000000,0.000000,-2.000000,1.000000,-2.000000,0.000000,0.000000,0.000000
25%,2017.000000,1.000000,0.000000,1.000000,0.000000,0.000000,0.000000,0.000000
50%,2019.000000,1.000000,0.000000,1.000000,0.000000,0.000000,1.000000,0.000000
75%,2021.000000,2.000000,0.000000,2.000000,0.000000,0.000000,1.000000,0.000000
max,2023.000000,750.000000,750.000000,1022.000000,1950.000000,94.000000,135.000000,250.000000


In [53]:
df.isnull().sum()

incident_type                             0
incident_year                             0
reported_month                            0
region_of_origin                         22
region_of_incident                        0
country_of_origin                         8
number_of_dead                          550
minimum_estimated_number_of_missing       0
total_number_of_dead_and_missing          0
number_of_survivors                       0
number_of_females                         0
number_of_males                           0
number_of_children                        0
cause_of_death                            0
migration_route                        3021
location_of_death                         0
information_source                        8
coordinates                              36
unsd_geographical_grouping                1
dtype: int64

## Data Cleaning:
* Investigated region labels containing "(P)" and retained them in the dataset, as they appear to indicate presumed region/country of origin when exact origin information was uncertain.
* Standardized entries within the Information Source column to correct spelling inconsistencies and merge duplicate categories.
* Investigated 654 duplicate records and retained them because they appeared to represent separate observations rather than obvious data-entry errors.
* Replaced missing values in categorical variables such as Region of Origin, Country of Origin, Migration Route, Information Source, and UNSD Geographical Grouping with "Unknown" where appropriate.
* Retained missing values (NaN) in numerical and geographic fields where information was unavailable, including Number of Dead and Coordinates.

In [54]:
df['region_of_origin'].value_counts()

region_of_origin
Latin America / Caribbean (P)    2164
Southern Asia                    1904
Unknown                          1737
Central America                  1565
Sub-Saharan Africa (P)           1528
Eastern Africa (P)               1133
Northern Africa                   452
Western Asia                      432
South America                     322
Eastern Africa                    298
Caribbean                         278
Western / Southern Asia (P)       245
Western Africa                    229
Northern Africa (P)               122
Sub-Saharan Africa                116
Mixed                             111
Southern Asia (P)                  90
South-eastern Asia                 68
Western Africa (P)                 52
Middle Africa                      51
South America (P)                  18
Europe                             14
Western / Southern Asia            14
Western Asia (P)                   12
Eastern Asia                       11
Caribbean (P)                    

In [55]:
df[df['region_of_origin'].isnull()]

,incident_type,incident_year,reported_month,region_of_origin,region_of_incident,country_of_origin,number_of_dead,minimum_estimated_number_of_missing,total_number_of_dead_and_missing,number_of_survivors,number_of_females,number_of_males,number_of_children,cause_of_death,migration_route,location_of_death,information_source,coordinates,unsd_geographical_grouping
1217,Incident,2016,January,NaN,Mediterranean,"Afghanistan,Iraq,Syrian Arab Republic",39.0,0,39,75,0,0,5,Drowning,Eastern Mediterranean,"between Ayvacik, Canakkale, Türkiye and Lesvos...",Turkish Coast Guard via IOM Athens. AFP and th...,"39.2893824, 26.4734281",Uncategorized
12923,Incident,2023,January,NaN,Northern Africa,Sudan,1.0,0,1,0,0,1,0,Sickness / lack of access to adequate healthcare,Sahara Desert crossing,"Al-Jawf, AlKufra, Libya",IOM Libya Displacement Tracking Matrix,"24.22092653, 23.291791",Northern Africa
12924,Incident,2023,January,NaN,Northern Africa,Egypt,1.0,0,1,0,0,1,0,Violence,Sahara Desert crossing,"Qondofah, Benghazi, Libya",IOM Libya Displacement Tracking Matrix,"32.029039, 20.016156",Northern Africa
12925,Incident,2023,January,NaN,Northern Africa,Unknown,1.0,0,1,0,0,1,0,Sickness / lack of access to adequate healthcare,Sahara Desert crossing,"Al-Madina, Tobruk, Libya",IOM Libya Displacement Tracking Matrix,"32.09704552, 23.991754",Northern Africa
12926,Incident,2023,February,NaN,Northern Africa,Nigeria,1.0,0,1,0,0,1,0,Vehicle accident / death linked to hazardous t...,NaN,"Rass Al-Magil, Misrata, Libya",IOM Libya Displacement Tracking Matrix,"32.33801861, 15.044267",Northern Africa
12927,Incident,2023,February,NaN,Northern Africa,Sudan,1.0,0,1,0,0,1,0,Accidental death,Sahara Desert crossing,"Al-Jawf, AlKufra, Libya",IOM Libya Displacement Tracking Matrix,"24.20637163, 23.290346",Northern Africa
12928,Incident,2023,January,NaN,Northern Africa,Unknown,3.0,0,3,4,0,7,0,Vehicle accident / death linked to hazardous t...,Sahara Desert crossing,"Alchuirif El gharbia, Ashshwayrif, Libya",IOM Libya Displacement Tracking Matrix,"31.04216493, 14.431426",Northern Africa
12929,Incident,2023,February,NaN,Northern Africa,Unknown,3.0,0,3,0,1,2,0,Mixed or unknown,Sahara Desert crossing,"Al-Jawf, AlKufra, Libya",IOM Libya Displacement Tracking Matrix,"24.20642056, 23.290271",Northern Africa
12930,Incident,2023,February,NaN,Northern Africa,Unknown,1.0,0,1,0,0,0,0,Harsh environmental conditions / lack of adequ...,Sahara Desert crossing,"Al-Jawf, AlKufra, Libya",IOM Libya Displacement Tracking Matrix,"24.20642056, 23.290271",Northern Africa
12931,Incident,2023,March,NaN,Northern Africa,Chad,1.0,0,1,0,0,1,0,Violence,Sahara Desert crossing,"Gatroun, Al-Qatroun, Libya",IOM Libya Displacement Tracking Matrix,"24.94447643, 14.627779",Northern Africa


In [56]:
df['region_of_origin'].value_counts()

region_of_origin
Latin America / Caribbean (P)    2164
Southern Asia                    1904
Unknown                          1737
Central America                  1565
Sub-Saharan Africa (P)           1528
Eastern Africa (P)               1133
Northern Africa                   452
Western Asia                      432
South America                     322
Eastern Africa                    298
Caribbean                         278
Western / Southern Asia (P)       245
Western Africa                    229
Northern Africa (P)               122
Sub-Saharan Africa                116
Mixed                             111
Southern Asia (P)                  90
South-eastern Asia                 68
Western Africa (P)                 52
Middle Africa                      51
South America (P)                  18
Europe                             14
Western / Southern Asia            14
Western Asia (P)                   12
Eastern Asia                       11
Caribbean (P)                    

In [57]:
df['region_of_origin'] = df['region_of_origin'].fillna('Unknown')

In [58]:
df[df['country_of_origin'].isnull()].head(10)

,incident_type,incident_year,reported_month,region_of_origin,region_of_incident,country_of_origin,number_of_dead,minimum_estimated_number_of_missing,total_number_of_dead_and_missing,number_of_survivors,number_of_females,number_of_males,number_of_children,cause_of_death,migration_route,location_of_death,information_source,coordinates,unsd_geographical_grouping
132,Incident,2014,July,Unknown,Mediterranean,NaN,13.0,0,13,0,0,0,0,Mixed or unknown,Central Mediterranean,Boat drifting in Libyan waters,IANS,"33.0335, 15",Uncategorized
185,Incident,2014,August,Unknown,Mediterranean,NaN,0.0,100,100,0,0,0,0,Drowning,Central Mediterranean,"Off the shore of al-Qarbouli, Libya",Press Association,"32.8879, 14",Uncategorized
254,Cumulative Incident,2014,September,Unknown,Eastern Africa,NaN,16.0,0,16,0,0,0,0,Mixed or unknown,Horn of Africa to Yemen crossing,Multiple locations,IOM Djibouti,"9.1303782, 41.2808577",Uncategorized
423,Incident,2015,May,South-eastern Asia,South-eastern Asia,NaN,2.0,0,2,0,0,0,0,Mixed or unknown,NaN,"Trafficking camps in Phang Nga province, Thailand",The Guardian,"8.4501414, 98.525532",South-eastern Asia
761,Incident,2015,October,Latin America / Caribbean (P),Central America,NaN,1.0,0,1,0,0,1,0,Violence,NaN,"Xaloztoc, Tlaxcala, Mexico",Agenda Tlaxcala,"19.408, -98.052",Central America
922,Incident,2015,November,Central America,North America,NaN,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,"31.511, -111.561833",Northern America
974,Incident,2015,December,Central America,North America,NaN,1.0,0,1,0,0,1,0,Drowning,US-Mexico border crossing,"All American Canal, CA near Calexico",El Mexicano,"32.668019, -115.502043",Northern America
1003,Incident,2015,December,Eastern Africa (P),Northern Africa,NaN,2.0,0,2,0,0,0,0,Sickness / lack of access to adequate healthcare,Sahara Desert crossing,"Sabha, Libya",Mixed Migration Monitoring Mechanism Initative...,"27.032288, 14.456116",Northern Africa


In [59]:
df['country_of_origin'] = df['country_of_origin'].fillna('Unknown')

In [60]:
df['country_of_origin'].value_counts().tail(55)

country_of_origin
Egypt,Eritrea,Sudan,Syrian Arab Republic                           1
Libya,Morocco,Syrian Arab Republic                                 1
Ecuador,Guatemala                                                  1
Myanmar,Viet Nam                                                   1
Angola,Cuba                                                        1
Côte d'Ivoire,Mali,Senegal,Unknown                                 1
Bangladesh,Egypt,Morocco,Unknown                                   1
Guinea,Senegal                                                     1
Somalia,State of Palestine,Syrian Arab Republic,Yemen              1
India,Pakistan                                                     1
Kenya                                                              1
Eritrea,Sudan                                                      1
Lao People's Democratic Republic                                   1
Egypt,Pakistan                                                     1
Morocco,Somalia,

In [61]:
df[df['number_of_dead'].isnull()]

,incident_type,incident_year,reported_month,region_of_origin,region_of_incident,country_of_origin,number_of_dead,minimum_estimated_number_of_missing,total_number_of_dead_and_missing,number_of_survivors,number_of_females,number_of_males,number_of_children,cause_of_death,migration_route,location_of_death,information_source,coordinates,unsd_geographical_grouping
295,Split Incident,2014,October,Western Asia,Mediterranean,"State of Palestine,Sudan",NaN,6,6,20,0,3,0,Drowning,Eastern Mediterranean,<p>Unspecified location in the Aegean Sea betw...,rd.nl,"38.74477, 26.486067",Uncategorized
336,Incident,2015,February,Western Africa,Mediterranean,"Côte d'Ivoire,Gambia,Guinea,Mali,Mauritania,Ni...",NaN,307,307,9,0,0,0,Drowning,Central Mediterranean,Between Libya and Lampedusa,IOM Italy from Flavio 29/04/2015,"34.3616, 12.9858",Uncategorized
351,Incident,2015,March,Sub-Saharan Africa (P),Western Africa,Unknown,NaN,25,25,0,0,0,0,Drowning,Western Africa / Atlantic route to the Canary ...,"Off the coast of Nuadibu, Mauritania",Asociación Pro Derechos Humanos de Andalucía,"20.94764809, -17.087039",Uncategorized
378,Cumulative Incident,2015,March,South-eastern Asia,South-eastern Asia,Myanmar,NaN,243,243,0,0,0,0,Mixed or unknown,NaN,Bay of Bengal,UNHCR,"17.64473, 92.46209",South-eastern Asia
379,Cumulative Incident,2015,March,South-eastern Asia,South-eastern Asia,Myanmar,NaN,130,130,0,0,0,0,Drowning,NaN,"Bay of Bengal - departed from Myanmar, Rakhine...",Arakan Project,"13.498141, 89.487406",South-eastern Asia
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
12946,Incident,2023,July,Unknown,Northern Africa,Unknown,NaN,1,1,21,0,0,0,Mixed or unknown,NaN,"Survivors rescued from Tawil Rutba, near Al-As...",IOM Libya,"32.80827441, 11.584487",Northern Africa
12950,Incident,2023,July,Unknown,Northern Africa,Unknown,NaN,2,2,12,0,0,0,Mixed or unknown,NaN,"Survivors rescued from Tawil Rutba, near Al-As...",IOM Libya,"32.80827441, 11.584487",Northern Africa
12966,Incident,2023,July,Unknown,Mediterranean,Unknown,NaN,4,4,13,0,0,0,Drowning,Western Mediterranean,"Unspecified location ""far"" from the coast of A...",Heroes del Mar,"36.3423573, -1.783771",Uncategorized
12967,Split Incident,2023,July,Western Asia,Mediterranean,"Algeria,Syrian Arab Republic",NaN,10,10,1,1,7,2,Drowning,Western Mediterranean,Unspecified location in international waters a...,Heroes del Mar,"36.09861941, -1.432208",Uncategorized


In [62]:
df['number_of_dead'] = df['number_of_dead'].fillna('NaN')

In [63]:
df[df['migration_route'].isnull()]

,incident_type,incident_year,reported_month,region_of_origin,region_of_incident,country_of_origin,number_of_dead,minimum_estimated_number_of_missing,total_number_of_dead_and_missing,number_of_survivors,number_of_females,number_of_males,number_of_children,cause_of_death,migration_route,location_of_death,information_source,coordinates,unsd_geographical_grouping
4,Incident,2014,January,Northern Africa,Europe,Sudan,1.0,0,1,2,0,1,0,Harsh environmental conditions / lack of adequ...,NaN,Border between Russia and Estonia,EUBusiness (Agence France-Presse),"59.1551, 28",Northern Europe
16,Cumulative Incident,2014,February,South-eastern Asia,South-eastern Asia,Myanmar,5.0,0,5,500,0,5,0,Sickness / lack of access to adequate healthcare,NaN,"Hat Yai hospital, 182 Ratthakan, Hat Yai, Hat ...",Manager Online,"7.0167471, 100.4674982",South-eastern Asia
28,Incident,2014,February,Southern Asia,Southern Asia,Bangladesh,1.0,0,1,0,0,1,0,Violence,NaN,India,Daily Star,"25.1841, 89",Southern Asia
33,Incident,2014,March,Southern Asia,South-eastern Asia,Cambodia,5.0,0,5,4,2,3,0,Vehicle accident / death linked to hazardous t...,NaN,"Bypass Road, Moo 7, Bang Phra subdistrict, Sri...","Manager Online, INN News, Khomchudluek","13.2353642, 101",South-eastern Asia
34,Incident,2014,March,Southern Asia,South-eastern Asia,Myanmar,2.0,0,2,10,0,2,0,Vehicle accident / death linked to hazardous t...,NaN,"Moo 13, 38th km of the 4006 Langsuan-Pato Road...","INN News, Manager Online, Daily News","9.96833, 99",South-eastern Asia
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
13004,Incident,2023,July,Unknown,Northern Africa,Unknown,1.0,0,1,0,1,0,0,Mixed or unknown,NaN,"Body recovered near Tawil Rutba, Libya, presum...",IOM Libya,"32.80827441, 11.584487",Northern Africa
13005,Incident,2023,July,Western Africa,Northern Africa,Sierra Leone,1.0,0,1,0,1,0,0,Mixed or unknown,NaN,"Body recovered near Tawil Rutba, Libya, presum...",IOM Libya,"32.80827441, 11.584487",Northern Africa
13006,Incident,2023,July,Unknown,Northern Africa,Unknown,1.0,0,1,10,0,1,0,Mixed or unknown,NaN,"Body recovered near Tawil Rutba, Libya, presum...",IOM Libya,"32.80827441, 11.584487",Northern Africa
13011,Incident,2023,July,Central America,North America,Mexico,NaN,1,1,0,0,1,0,Mixed or unknown,NaN,"Fruit Stand & Orchards, Osooyos, British Colum...","Gendarmerie royale du Canada, El Mañana","49.02831207, -119.432156",Northern Africa


In [64]:
df['migration_route'].value_counts()

migration_route
US-Mexico border crossing                                3392
Sahara Desert crossing                                   2046
Central Mediterranean                                    1106
Afghanistan to Iran                                      1099
Western Mediterranean                                     614
Eastern Mediterranean                                     336
Western Africa / Atlantic route to the Canary Islands     226
Western Balkans                                           210
Horn of Africa to Yemen crossing                          161
Türkiye-Europe land route                                 157
English Channel to the UK                                 134
Syria to Türkiye                                          129
Darien                                                     98
Caribbean to US                                            63
Belarus-EU border                                          61
Dominican Republic to Puerto Rico                     

In [65]:
df['migration_route'] = df['migration_route'].fillna('Unknown')

In [66]:
df[df['information_source'].isnull()]

,incident_type,incident_year,reported_month,region_of_origin,region_of_incident,country_of_origin,number_of_dead,minimum_estimated_number_of_missing,total_number_of_dead_and_missing,number_of_survivors,number_of_females,number_of_males,number_of_children,cause_of_death,migration_route,location_of_death,information_source,coordinates,unsd_geographical_grouping
62,Incident,2014,April,Western Asia,Mediterranean,Syrian Arab Republic,8.0,1,9,5,2,0,1,Drowning,Eastern Mediterranean,Türkiye waters - en route to Greece,NaN,"38.669208, 27",Uncategorized
465,Incident,2015,June,Latin America / Caribbean (P),Central America,Unknown,3.0,0,3,117,0,0,0,Violence,Unknown,"Sonora, Mexico",NaN,"29.3766625, -113.982346",Central America
545,Incident,2015,July,Latin America / Caribbean (P),North America,Unknown,2.0,0,2,0,0,0,0,Mixed or unknown,US-Mexico border crossing,"Desert near Carrizo Springs, Dimmit County, Te...",NaN,"28.2597513, -100.245529",Northern America
724,Incident,2015,September,Central America,Central America,"Guatemala,Honduras",2.0,0,2,0,0,0,0,Vehicle accident / death linked to hazardous t...,Unknown,"Puebla, Mexico",NaN,"19.049997, -98.199999",Central America
3142,Incident,2017,August,Unknown,South-eastern Asia,Unknown,2.0,0,2,0,0,2,0,Harsh environmental conditions / lack of adequ...,Unknown,"Kuala Lumpur, Malaysia",NaN,"3.233046, 101.610573",South-eastern Asia
5360,Incident,2018,February,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,NaN,"33.061615, -113.367348",NaN
6987,Incident,2019,November,Eastern Africa,Western Asia,Ethiopia,26.0,0,26,74,8,18,0,Violence,Unknown,"Near Al-Raqw Market, Monabbih, Saada province,...",NaN,"17.245130865, 43.239432135",Western Asia
8176,Incident,2020,December,Northern Africa (P),Mediterranean,Unknown,NaN,5,5,0,0,5,0,Drowning,Western Mediterranean,Unspecified location in the Alboran Sea. Depar...,NaN,"35.30420753, -3.92007383",Uncategorized


In [67]:
df['information_source'] = df['information_source'].fillna('Unknown')

In [68]:
df['information_source'].value_counts().head(55)

information_source
IOM Afghanistan                                          1538
Pima County Office of the Medical Examiner (PCOME)       1480
Mixed Migration Monitoring Mechanism Initiative (4Mi)    1089
Mixed Migration Monitoring Mechanism Initative (4mi)      992
Mixed Migration Monitoring Mechanism Initiative (4mi)     673
IOM Libya                                                 354
MHub                                                      182
US Border Patrol                                          132
Syrian Observatory for Human Rights                        99
Zócalo                                                     96
IOM Italy                                                  74
UNHCR                                                      73
IOM Libya Displacement Tracking Matrix                     71
OHCHR Special Procedures letter                            68
Imperial County Coroner's Office                           62
IOM Djibouti                                       

In [69]:
df['information_source'] = df['information_source'].replace({'Mixed Migration Monitoring Mechanism Initative (4mi)':'Mixed Migration Monitoring Mechanism Initiative (4Mi)','Mixed Migration Monitoring Mechanism Initiative (4mi)':'Mixed Migration Monitoring Mechanism Initiative (4Mi)'})

In [70]:
df['information_source'].value_counts().head(55)

information_source
Mixed Migration Monitoring Mechanism Initiative (4Mi)    2754
IOM Afghanistan                                          1538
Pima County Office of the Medical Examiner (PCOME)       1480
IOM Libya                                                 354
MHub                                                      182
US Border Patrol                                          132
Syrian Observatory for Human Rights                        99
Zócalo                                                     96
IOM Italy                                                  74
UNHCR                                                      73
IOM Libya Displacement Tracking Matrix                     71
OHCHR Special Procedures letter                            68
Imperial County Coroner's Office                           62
IOM Djibouti                                               48
South Texas Human Rights Center                            46
Hidalgo County Sheriff's Office                    

In [71]:
df[df['coordinates'].isnull()]

,incident_type,incident_year,reported_month,region_of_origin,region_of_incident,country_of_origin,number_of_dead,minimum_estimated_number_of_missing,total_number_of_dead_and_missing,number_of_survivors,number_of_females,number_of_males,number_of_children,cause_of_death,migration_route,location_of_death,information_source,coordinates,unsd_geographical_grouping
231,Incident,2014,October,Unknown,Mediterranean,Unknown,0.0,20,20,93,0,0,0,Drowning,Central Mediterranean,off the coast of Libya,Mail Online and IOM Rome,NaN,Uncategorized
256,Incident,2014,January,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
257,Incident,2014,February,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Harsh environmental conditions / lack of adequ...,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
258,Incident,2014,March,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Violence,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
259,Incident,2014,April,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,0,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
260,Incident,2014,April,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Violence,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
261,Incident,2014,May,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
262,Incident,2014,June,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
263,Incident,2014,June,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
264,Incident,2014,July,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America


In [72]:
df['coordinates'] = df['coordinates'].fillna('NaN')

In [73]:
df[df['unsd_geographical_grouping'].isnull()]

,incident_type,incident_year,reported_month,region_of_origin,region_of_incident,country_of_origin,number_of_dead,minimum_estimated_number_of_missing,total_number_of_dead_and_missing,number_of_survivors,number_of_females,number_of_males,number_of_children,cause_of_death,migration_route,location_of_death,information_source,coordinates,unsd_geographical_grouping
5360,Incident,2018,February,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Unknown,"33.061615, -113.367348",NaN


In [74]:
df['unsd_geographical_grouping'] = df['unsd_geographical_grouping'].fillna('Unknown')

In [75]:
df['unsd_geographical_grouping'].value_counts()

unsd_geographical_grouping
Northern America      2708
Uncategorized         2351
Northern Africa       1872
Southern Asia         1660
Central America       1362
Western Africa         941
Eastern Africa         467
Western Asia           389
Southern Europe        329
South-eastern Asia     225
South America          207
Western Europe         156
Caribbean              113
Eastern Europe         111
Middle Africa           75
Northern Europe         25
Eastern Asia            15
Southern Africa         12
Central Asia             1
Unknown                  1
Name: count, dtype: int64

In [76]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13020 entries, 0 to 13019
Data columns (total 19 columns):
 #   Column                               Non-Null Count  Dtype 
---  ------                               --------------  ----- 
 0   incident_type                        13020 non-null  object
 1   incident_year                        13020 non-null  int64 
 2   reported_month                       13020 non-null  object
 3   region_of_origin                     13020 non-null  object
 4   region_of_incident                   13020 non-null  object
 5   country_of_origin                    13020 non-null  object
 6   number_of_dead                       13020 non-null  object
 7   minimum_estimated_number_of_missing  13020 non-null  int64 
 8   total_number_of_dead_and_missing     13020 non-null  int64 
 9   number_of_survivors                  13020 non-null  int64 
 10  number_of_females                    13020 non-null  int64 
 11  number_of_males                      1302

In [77]:
df.duplicated().sum()

np.int64(641)

In [78]:
df[df.duplicated()].head(45)

,incident_type,incident_year,reported_month,region_of_origin,region_of_incident,country_of_origin,number_of_dead,minimum_estimated_number_of_missing,total_number_of_dead_and_missing,number_of_survivors,number_of_females,number_of_males,number_of_children,cause_of_death,migration_route,location_of_death,information_source,coordinates,unsd_geographical_grouping
222,Incident,2014,October,Eastern Africa,Eastern Africa,Unknown,1.0,0,1,0,0,1,0,Drowning,Horn of Africa to Yemen crossing,Between Bosaso and Yemen (Bin Sinaa),SHS,"12.5331, 48",Uncategorized
223,Incident,2014,October,Eastern Africa,Eastern Africa,Unknown,1.0,0,1,0,0,1,0,Drowning,Horn of Africa to Yemen crossing,Between Bosaso and Yemen (Bin Sinaa),SHS,"12.5331, 48",Uncategorized
224,Incident,2014,October,Eastern Africa,Eastern Africa,Unknown,1.0,0,1,0,0,1,0,Drowning,Horn of Africa to Yemen crossing,Between Bosaso and Yemen (Bin Sinaa),SHS,"12.5331, 48",Uncategorized
263,Incident,2014,June,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
270,Incident,2014,October,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
271,Incident,2014,October,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
272,Incident,2014,October,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
273,Incident,2014,October,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
275,Incident,2014,October,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America
276,Incident,2014,October,Latin America / Caribbean (P),North America,Unknown,1.0,0,1,0,0,1,0,Mixed or unknown,US-Mexico border crossing,Pima Country Office of the Medical Examiner ju...,Pima County Office of the Medical Examiner (PC...,NaN,Northern America


Notes: Duplicates remained as the data sometimes has different coordinates, different numbers of people identified, and sometimes seems to be separate inputs for each individual. Information source for mixed migration monitoring mechanism initiative.

## Exploration Analysis

In [79]:
df['incident_type'].value_counts()

incident_type
Incident                   12670
Split Incident               261
Cumulative Incident           84
Incident,Split Incident        5
Name: count, dtype: int64

In [80]:
df['cause_of_death'].value_counts()

cause_of_death
Drowning                                                                                                                   3313
Mixed or unknown                                                                                                           3175
Vehicle accident / death linked to hazardous transport                                                                     2112
Harsh environmental conditions / lack of adequate shelter, food, water                                                     1360
Violence                                                                                                                   1313
Sickness / lack of access to adequate healthcare                                                                           1219
Accidental death                                                                                                            507
Drowning,Harsh environmental conditions / lack of adequate shelter, food, water          

In [81]:
df['cause_of_death'].value_counts(normalize=True)*100
#The number one leading cause of death (by a small margin) was drowning followed by mixed or unknown cause of death and this was followed by vehicle accident/death linked to hazardous transport.

cause_of_death
Drowning                                                                                                                   25.445469
Mixed or unknown                                                                                                           24.385561
Vehicle accident / death linked to hazardous transport                                                                     16.221198
Harsh environmental conditions / lack of adequate shelter, food, water                                                     10.445469
Violence                                                                                                                   10.084485
Sickness / lack of access to adequate healthcare                                                                            9.362519
Accidental death                                                                                                            3.894009
Drowning,Harsh environmental conditions / lack of adeq

In [82]:
df.groupby("region_of_incident")['cause_of_death'].value_counts(normalize=True)*100

region_of_incident  cause_of_death                                                        
Caribbean           Drowning                                                                  80.000000
                    Mixed or unknown                                                           6.875000
                    Violence                                                                   5.000000
                    Harsh environmental conditions / lack of adequate shelter, food, water     4.375000
                    Vehicle accident / death linked to hazardous transport                     3.125000
                                                                                                ...    
Western Asia        Vehicle accident / death linked to hazardous transport                     9.420290
                    Harsh environmental conditions / lack of adequate shelter, food, water     8.937198
                    Mixed or unknown                                         

In [83]:
df.groupby("region_of_incident")['cause_of_death'].value_counts()
result_df = df.groupby("region_of_incident")['cause_of_death'].value_counts().head(40).to_frame('Count').reset_index()
result_df

,region_of_incident,cause_of_death,Count
0,Caribbean,Drowning,128
1,Caribbean,Mixed or unknown,11
2,Caribbean,Violence,8
3,Caribbean,Harsh environmental conditions / lack of adequ...,7
4,Caribbean,Vehicle accident / death linked to hazardous t...,5
5,Caribbean,Accidental death,1
6,Central America,Drowning,591
7,Central America,Vehicle accident / death linked to hazardous t...,339
8,Central America,Violence,151
9,Central America,Mixed or unknown,132


In [84]:
df[['number_of_dead', 'number_of_females', 'number_of_males', 'number_of_children']] = df[['number_of_dead', 'number_of_females', 'number_of_males', 'number_of_children']].apply(pd.to_numeric, errors='coerce')
df.groupby("incident_year")[[
    'number_of_dead',
    'number_of_females',
    'number_of_males',
    'number_of_children'
]].sum()
#Number of dead and number of identified persons don't match, so not able to pull much about the persons per year.

,number_of_dead,number_of_females,number_of_males,number_of_children
incident_year,,,,
2014,1753.0,67,263,302
2015,4094.0,232,821,305
2016,4294.0,375,997,281
2017,3726.0,547,1606,315
2018,3203.0,1975,1771,363
2019,3622.0,887,1810,537
2020,2354.0,341,1585,159
2021,3651.0,719,2438,336
2022,4296.0,659,2623,369


In [85]:
df[['number_of_dead', 'number_of_females', 'number_of_males', 'number_of_children','number_of_survivors']] = df[['number_of_dead', 'number_of_females', 'number_of_males', 'number_of_children','number_of_survivors']].apply(pd.to_numeric, errors='coerce')
df.groupby("region_of_incident")[[
    'number_of_dead',
    'number_of_females',
    'number_of_males',
    'number_of_children',
    'number_of_survivors'
]].sum().sort_values(by='number_of_dead', ascending=False).head(45)

,number_of_dead,number_of_females,number_of_males,number_of_children,number_of_survivors
region_of_incident,,,,,
Mediterranean,9703.0,2600,3807,1104,57411
Northern Africa,5702.0,974,1669,410,6793
North America,3707.0,331,1983,64,934
Southern Asia,2272.0,172,1497,201,1076
Western Africa,2180.0,443,951,251,2360
Central America,2049.0,288,1310,179,3091
Western Asia,1931.0,327,1166,136,4163
Eastern Africa,1624.0,107,445,90,1658
South-eastern Asia,1321.0,325,314,440,4641


In [86]:
df.groupby("incident_year")['cause_of_death'].value_counts().head(40)
result_df = df.groupby("incident_year")['cause_of_death'].value_counts().head(40).to_frame('Count').reset_index()
result_df

,incident_year,cause_of_death,Count
0,2014,Mixed or unknown,138
1,2014,Drowning,83
2,2014,Vehicle accident / death linked to hazardous t...,25
3,2014,Harsh environmental conditions / lack of adequ...,23
4,2014,Violence,22
5,2014,Accidental death,3
6,2014,"Drowning,Mixed or unknown",1
7,2014,Sickness / lack of access to adequate healthcare,1
8,2015,Drowning,238
9,2015,Mixed or unknown,234


In [87]:
df.groupby("region_of_incident")['migration_route'].value_counts(ascending=False)

region_of_incident  migration_route                                      
Caribbean           Caribbean to US                                            54
                    Dominican Republic to Puerto Rico                          42
                    Unknown                                                    32
                    Haiti to Dominican Republic                                17
                    Venezuela to Caribbean                                     10
                    Western Africa / Atlantic route to the Canary Islands       3
                    Caribbean to Central America                                1
                    US-Mexico border crossing                                   1
Central America     US-Mexico border crossing                                 705
                    Unknown                                                   581
                    Darien                                                     88
                    West

In [88]:
df["migration_route"].value_counts()

migration_route
US-Mexico border crossing                                3392
Unknown                                                  3021
Sahara Desert crossing                                   2046
Central Mediterranean                                    1106
Afghanistan to Iran                                      1099
Western Mediterranean                                     614
Eastern Mediterranean                                     336
Western Africa / Atlantic route to the Canary Islands     226
Western Balkans                                           210
Horn of Africa to Yemen crossing                          161
Türkiye-Europe land route                                 157
English Channel to the UK                                 134
Syria to Türkiye                                          129
Darien                                                     98
Caribbean to US                                            63
Belarus-EU border                                     

In [89]:
df.groupby("migration_route")['number_of_dead'].sum().sort_values(ascending=False)

migration_route
Unknown                                                  7412.0
Central Mediterranean                                    7114.0
Sahara Desert crossing                                   5617.0
US-Mexico border crossing                                4512.0
Eastern Mediterranean                                    1548.0
Horn of Africa to Yemen crossing                         1512.0
Afghanistan to Iran                                      1234.0
Western Mediterranean                                    1042.0
Western Africa / Atlantic route to the Canary Islands     867.0
Western Balkans                                           314.0
Syria to Türkiye                                          262.0
Türkiye-Europe land route                                 238.0
English Channel to the UK                                 204.0
Iran to Türkiye                                           189.0
Darien                                                    171.0
Caribbean to US         

In [90]:
df.to_csv('cleaned_missing_migrants.csv', index = False)

## SQL Analysis: 
This analysis explores the high risk regions, migration routes, most frequent incidents, and total deaths spanning 2014-2023.  

In [91]:
import sqlite3
conn = sqlite3.connect("MissingMigrants.db")

In [92]:
df.to_sql("MissingMigrants", conn, if_exists = "replace", index = False)

13020

### Number of Dead and Missing based on the Region of Incident ###

In [93]:
query = """
SELECT "region_of_incident", SUM("number_of_dead") as deaths, SUM(minimum_estimated_number_of_missing) AS est_missing
FROM MissingMigrants
GROUP BY region_of_incident
ORDER BY deaths DESC;

"""
result = pd.read_sql_query (query, conn)
result

,region_of_incident,deaths,est_missing
0,Mediterranean,9703.0,17972
1,Northern Africa,5702.0,2197
2,North America,3707.0,63
3,Southern Asia,2272.0,222
4,Western Africa,2180.0,585
5,Central America,2049.0,204
6,Western Asia,1931.0,211
7,Eastern Africa,1624.0,313
8,South-eastern Asia,1321.0,1793
9,Europe,962.0,57


The top 5 regions of incidents were: Mediterranean (9703), Northern Africa (5702), North America (3707), Southern Asia (2272), and Western Africa (2180). 

### Leading Cause Of Death

In [94]:
query = """
SELECT cause_of_death, COUNT("cause_of_death") AS value_count, SUM("number_of_dead") AS deaths, SUM(minimum_estimated_number_of_missing) AS est_missing
FROM MissingMigrants
GROUP BY cause_of_death
ORDER BY value_count DESC;
"""
result = pd.read_sql_query (query, conn)
result

,cause_of_death,value_count,deaths,est_missing
0,Drowning,3313,12578.0,21769
1,Mixed or unknown,3175,6561.0,1965
2,Vehicle accident / death linked to hazardous t...,2112,4887.0,43
3,Harsh environmental conditions / lack of adequ...,1360,2827.0,541
4,Violence,1313,3446.0,141
5,Sickness / lack of access to adequate healthcare,1219,1525.0,48
6,Accidental death,507,697.0,2
7,"Drowning,Harsh environmental conditions / lack...",8,108.0,86
8,"Drowning,Mixed or unknown",4,50.0,181
9,Harsh environmental conditions / lack of adequ...,3,14.0,0


The leading cause of death is drowning followed my mixed or unknown causes of death.

In [98]:
df['cause_of_death'].value_counts()

cause_of_death
Drowning                                                                                                                   3313
Mixed or unknown                                                                                                           3175
Vehicle accident / death linked to hazardous transport                                                                     2112
Harsh environmental conditions / lack of adequate shelter, food, water                                                     1360
Violence                                                                                                                   1313
Sickness / lack of access to adequate healthcare                                                                           1219
Accidental death                                                                                                            507
Drowning,Harsh environmental conditions / lack of adequate shelter, food, water          

In [102]:
query = """
SELECT region_of_incident, cause_of_death, COUNT(*) AS incidents, SUM("number_of_dead") AS deaths, SUM(minimum_estimated_number_of_missing) AS est_missing
FROM MissingMigrants
GROUP BY region_of_incident, cause_of_death
ORDER BY region_of_incident, deaths DESC;
"""
result = pd.read_sql_query (query, conn)
result

,region_of_incident,cause_of_death,incidents,deaths,est_missing
0,Caribbean,Drowning,128,324.0,1002
1,Caribbean,Harsh environmental conditions / lack of adequ...,7,80.0,8
2,Caribbean,Mixed or unknown,11,25.0,39
3,Caribbean,Vehicle accident / death linked to hazardous t...,5,16.0,0
4,Caribbean,Violence,8,8.0,0
...,...,...,...,...,...
108,Western Asia,Vehicle accident / death linked to hazardous t...,39,180.0,0
109,Western Asia,Harsh environmental conditions / lack of adequ...,37,122.0,6
110,Western Asia,Mixed or unknown,15,31.0,21
111,Western Asia,Sickness / lack of access to adequate healthcare,6,12.0,0


### High Risk Migration Routes

In [101]:
query = """
SELECT
    migration_route,SUM(number_of_dead) AS deaths, SUM(minimum_estimated_number_of_missing) AS est_missing
FROM MissingMigrants
GROUP BY migration_route
ORDER BY deaths DESC;
"""
result = pd.read_sql_query (query, conn)
result

,migration_route,deaths,est_missing
0,Unknown,7412.0,2521
1,Central Mediterranean,7114.0,14902
2,Sahara Desert crossing,5617.0,156
3,US-Mexico border crossing,4512.0,88
4,Eastern Mediterranean,1548.0,745
5,Horn of Africa to Yemen crossing,1512.0,318
6,Afghanistan to Iran,1234.0,5
7,Western Mediterranean,1042.0,2325
8,Western Africa / Atlantic route to the Canary ...,867.0,2571
9,Western Balkans,314.0,29


The migration route with the highest deaths are Unknown (7412 deaths) followed by Central Mediterranean (7114), Sahara Desert Crossing (5617 deaths), and US-Mexico Border Crossing (4512 deaths). The number of missing migrants however is more frequently documented in the Central Mediterranean route with 14,902 missing.

In [100]:
query = """
SELECT region_of_incident, migration_route, COUNT(*) AS count, SUM("number_of_dead") AS deaths, SUM(minimum_estimated_number_of_missing) AS est_missing
FROM MissingMigrants
GROUP BY migration_route
ORDER BY region_of_incident, count DESC;
"""
result = pd.read_sql_query (query, conn)
result

,region_of_incident,migration_route,count,deaths,est_missing
0,Caribbean,Caribbean to US,63,109.0,390
1,Caribbean,Dominican Republic to Puerto Rico,42,77.0,228
2,Caribbean,Haiti to Dominican Republic,17,55.0,9
3,Caribbean,Venezuela to Caribbean,11,55.0,132
4,Caribbean,Caribbean to Central America,1,NaN,17
5,Eastern Africa,Horn of Africa to Yemen crossing,161,1512.0,318
6,Eastern Africa,Comoros to Mayotte,16,70.0,144
7,Europe,Unknown,3021,7412.0,2521
8,Europe,Western Balkans,210,314.0,29
9,Europe,English Channel to the UK,134,204.0,9


### Fatalities over Time

In [99]:
query = """
SELECT incident_year, SUM("number_of_dead") AS deaths, SUM(minimum_estimated_number_of_missing) AS est_missing
FROM MissingMigrants
GROUP BY incident_year
ORDER BY incident_year
"""
result = pd.read_sql_query (query, conn)
result

,incident_year,deaths,est_missing
0,2014,1753.0,3568
1,2015,4094.0,2655
2,2016,4294.0,3784
3,2017,3726.0,2564
4,2018,3203.0,1806
5,2019,3622.0,1796
6,2020,2354.0,1947
7,2021,3651.0,2500
8,2022,4296.0,2766
9,2023,1725.0,1409


Deaths and number of missing fluctuates througout the years with minimum at 1725 deaths and 1409 missing and maximums at 4296 deaths and 2766 missing.

Findings:
* The top 5 regions of incidents were: Mediterranean (9703), Northern Africa (5702), North America (3707), Southern Asia (2272), and Western Africa (2180).
* The main leading cause of death is drowning (12578) followed by unknown cause of death (6561) and Vehicle accident / death linked to hazardous transport (4887).
* The migration route with the highest deaths are Unknown (7412 deaths) followed by Central Mediterranean (7114), Sahara Desert Crossing (5617 deaths), and US-Mexico Border Crossing (4512 deaths). The number of missing migrants however is more frequently documented in the Central Mediterranean route with 14,902 missing.
* Deaths and number of missing fluctuates througout the years with the minimums recorded at 1725 deaths and 1409 missing and maximums recorded at 4296 deaths and 2766 missing.

Conclusion:
Across occupations, average sleep quality, stress scores, and work hours varied considerably. Occupations with higher average sleep quality generally reported lower average stress scores, with a moderate negative correlation observed between sleep quality and stress (r = -0.64). Analysis of wake episodes among lawyers showed that increased wake episodes were associated with lower sleep quality and slightly higher stress levels. Overall, the results suggest that sleep quality may be more closely associated with stress than work hours alone within this dataset. However, these findings represent associations rather than causal relationships, and other occupation-specific or individual factors may also contribute to differences in stress levels.